File: 240116/국영수 순이지/korean-english-math-order.py
n = int(input())

class exam:

    def __init__(self,name,korean,math,eng):
        self.name = name
        self.korean = korean 
        self.math = math
        self.eng = eng

student = [] 

for _ in range(n):
    name, korean, math, eng = map(str,input().split())
    korean = int(korean)
    math = int(math)
    eng = int(eng)
    student.append(exam(name,korean,math,eng))

student.sort(key = lambda x: (-x.korean,-x.math,-x.eng))

for i in range(n):
    print(student[i].name,student[i].korean,student[i].math,student[i].eng)

File: 240116/국영수 순이지/korean-english-math-order.java
import java.util.Scanner;
import java.util.Arrays;

class Student implements Comparable<Student>{
    String name ;
    int kor,math,eng;

    public Student(String name,int kor,int math,int eng){
        this.name = name;
        this.kor = kor;
        this.math = math;
        this.eng = eng;
    }

    public int compareTo(Student student){
        if (student.kor != this.kor){
            return student.kor - this.kor;
        }
        if (student.math!=this.math){
            return student.math - this.math;
        }
        return student.eng - this.eng;
    }
}
public class Main {
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        Student[] human = new Student[n];

        for (int i = 0 ; i<n; i++){
            String name = sc.next();
            int kor = sc.nextInt();
            int math = sc.nextInt();
            int eng = sc.nextInt();
            human[i] = new Student(name,kor,math,eng);
        }

        Arrays.sort(human);

        for (int i = 0 ; i<n; i++){
            System.out.println(human[i].name+" "+human[i].kor+" "+human[i].math+" "+human[i].eng);
        }
    }
}